File: src/ReadStorm.Domain/Models/ChapterStatus.cs
namespace ReadStorm.Domain.Models;

/// <summary>章节下载状态。</summary>
public enum ChapterStatus
{
    /// <summary>待下载。</summary>
    Pending = 0,

    /// <summary>下载中。</summary>
    Downloading = 1,

    /// <summary>已完成。</summary>
    Done = 2,

    /// <summary>下载失败。</summary>
    Failed = 3,
}

File: pureopus/app/src/main/java/com/readstorm/app/domain/models/ChapterStatus.kt
package com.readstorm.app.domain.models

enum class ChapterStatus(val value: Int) {
    Pending(0),
    Downloading(1),
    Done(2),
    Failed(3);

    companion object {
        fun fromValue(value: Int): ChapterStatus =
            entries.firstOrNull { it.value == value } ?: Pending
    }
}
